File: ModoMercado/Core/Domain/Models/Mercado.swift
import Foundation

struct Mercado: Identifiable, Equatable, Codable {
    let id: UUID
    var nome: String
    let criadoEm: Date

    init(
        id: UUID = UUID(),
        nome: String,
        criadoEm: Date = Date()
    ) {
        self.id = id
        self.nome = nome
        self.criadoEm = criadoEm
    }
}

File: modo_mercado/lib/core/domain/models/mercado.dart
import 'package:equatable/equatable.dart';

class Mercado extends Equatable {
  final String id;
  final String nome;
  final DateTime criadoEm;

  const Mercado({
    required this.id,
    required this.nome,
    required this.criadoEm,
  });

  Mercado copyWith({String? id, String? nome, DateTime? criadoEm}) {
    return Mercado(
      id: id ?? this.id,
      nome: nome ?? this.nome,
      criadoEm: criadoEm ?? this.criadoEm,
    );
  }

  @override
  List<Object?> get props => [id, nome, criadoEm];
}
